### ISSUE TITLE
Invalid assignment passing type checking for imported types

### ISSUE THREAD

**Issue by DamonOehlman** [facebook/flow#3065]
This is quite possibly user error (given I couldn't find _any_ other issues on this) but I'm seeing assignment of member properties accept invalid values (based on the type definition) when a type is imported from another file.

Here's the breaking case (where validation passes, and it shouldn't):

```js
// @flow

import type { Rect } from './primitives.js';

type LayerOptions = {
  size: Rect
};

export class Layer {
  size: Rect;

  constructor(options: LayerOptions) {
    this.size = options.size || { x: 0, y: 0, width: 0, height: 0 };
  }

  updateSize(rect: Rect) {
    this.size = 'asdfasdf';
    this.size = 5; // Object.assign(rect);
  }
}
```

And the passing case (validation fails, type defined in the same file):

```js
// @flow

type Rect = {|
  x: number,
  y: number,
  width: number,
  height: number
|};

type LayerOptions = {
  size: Rect
};

export class Layer {
  size: Rect;

  constructor(options: LayerOptions) {
    this.size = options.size || { x: 0, y: 0, width: 0, height: 0 };
  }

  updateSize(rect: Rect) {
    this.size = 'asdfasdf';
    this.size = 5; // Object.assign(rect);
  }
}
```

Any help on this would be greatly appreciated!

**Comment by gcanti** [comment]
@DamonOehlman I can see the error also in the first case, are you using Nuclide or running flow from the command line?

**Comment by DamonOehlman** [comment]
@gcanti I'm seeing it both Nuclide and the command line...

**Comment by billschaller** [comment]
This seems to be resolved as of 0.41

**Comment by DamonOehlman** [comment]
@billschaller Yep - I'm seeing the problem resolved in `0.41` - thanks for the heads up.  Closing the issue :)
